R/generateHypotheses.R

```diff
@@ -1,192 +1,305 @@
-  hypothesis[,consecutive_feature_identifier := .I]
-  hypothesis[,unique_feature_identifier := 0]
-  unique_feature_identifier=0
-  tree_features <- names(tree_cut)
-  unique_tree_groups <- unique(tree_cut)
-  for (tree_group in unique_tree_groups) {
-    unique_feature_identifier = unique_feature_identifier+1
-    tree_complex_ids <- which(tree_cut==tree_group)
-    hypothesis$unique_feature_identifier[tree_complex_ids] = unique_feature_identifier
-  }
-  hypothesis <- hypothesis[order(unique_feature_identifier,-n_subunits)]
-  hypothesis[,complex_name := paste(complex_name,collapse="; "),by="unique_feature_identifier"]
-  hypothesis_unique <- unique(hypothesis,by="unique_feature_identifier")
-  hypothesis_unique <- subset(hypothesis_unique,select=c("complex_id","complex_name","subunits_detected"))
-  hypothesis_unique <- hypothesis_unique[,list(protein_id = unlist(strsplit(subunits_detected, ";"))), by=c("complex_id","complex_name")]
-  hypothesis_unique
+  
+  hypothesis_unique <- .collapseWideHypothesis(hypothesis)
+  return(hypothesis_unique)
 }
 
 
 
-#' generateDecoys
+#' Generate one comlpex decoy-hypothesis
 #' @description Generate list of proteins for one complex decoy hypothesis
 #' @import data.table
-#' @param size numeric number of proteins to generate for complex decoy hypothesis
-#' @param all_proteins character vector of all possible protein_ids
+#' @param size Numeric, number of proteins to generate for complex decoy hypothesis
+#' @param all_proteins Character, vector of all possible protein_ids
 #' @param dist_info data.table with three columns (x,y,dist) containing all possible combinations
 #' of proteins and their shortest distance.
 #' This is the output from calculatePathlength()
-#' @param min_distance numeric minimum distance between to proteins to be able to be in same complex decoy
-#' @return data.table with binary interactions between "a" and "b"
-#' @export
-generateDecoys <- function(size,all_proteins,dist_info,min_distance){
-  complex_proteins <- vector(mode="character",length=size)
-  for(j in c(1:size)){
-    combi_sub <- subset(dist_info,(((x %in% complex_proteins) | (y %in% complex_proteins)) & (dist<min_distance)))
-    impossible_proteins <- unique(c(combi_sub$x,combi_sub$y))
-    possible_proteins <- all_proteins[! all_proteins %in% impossible_proteins]
-    if (length(possible_proteins) > 0){
-      complex_proteins[j] <- sample(possible_proteins,1)
-    } else {
-      stop(paste0("Network connectivity is too high for decoy generation at a min_distance of ",min_distance, ". Please select a lower min_distance cutoff."))
+#' @param min_distance Numeric, minimum distance between to proteins to be
+#' able to be in same complex decoy
+#' @return character with a complex hypothesis, protein ids are separated by ';'
+#' 
+generateDecoys <- function(size,
+                           all_proteins,
+                           dist_info,
+                           min_distance = 2,
```

Comment:
Maybe we should set this to 1, because if no subsetted dist_info is provided, the min_distance is not used!
In the current state of the function, the min_distance is only used for the error messages